java/datastructures/lib/src/test/java/datastructures/graphs/TestGraph.java

```diff
@@ -82,5 +91,43 @@ public class TestGraph {
     Assertions.assertEquals(vertexA, neighbors.getFirst().destination);
     Assertions.assertEquals(2, neighbors.getFirst().weight);
   }
+
+  @Test public void testDepthFirstTraversal() {
+    Graph<Character> graph = new Graph<>(6);
+    Vertex<Character> nullV = new Vertex<>(null);
+    Vertex<Character> a = graph.addVertex('A');
+    Vertex<Character> b = graph.addVertex('B');
+    Vertex<Character> c = graph.addVertex('C');
+    Vertex<Character> d = graph.addVertex('D');
+    Vertex<Character> e = graph.addVertex('E');
+    Vertex<Character> f = graph.addVertex('F');
+    Vertex<Character> g = graph.addVertex('G');
+    Vertex<Character> h = graph.addVertex('H');
+
+    graph.addEdge(a, b);
+    graph.addEdge(a, d);
+    graph.addEdge(b, c);
+    graph.addEdge(b, d);
+    graph.addEdge(c, g);
+    graph.addEdge(d, e);
+    graph.addEdge(d, h);
+    graph.addEdge(d, f);
+    graph.addEdge(f, h);
+
+    LinkedList<Vertex<Character>> expectedTraversalOrder = new LinkedList<>();
+    expectedTraversalOrder.add(a);
+    expectedTraversalOrder.add(b);
+    expectedTraversalOrder.add(d);
+    expectedTraversalOrder.add(e);
+    expectedTraversalOrder.add(c);
+    expectedTraversalOrder.add(f);
+    expectedTraversalOrder.add(g);
+    expectedTraversalOrder.add(h);
+
+    LinkedList<Vertex<Character>> traversalOrder = graph.depthFirstTraversal(a);
+
+//    Assertions.assertEquals(expectedTraversalOrder, traversalOrder);
+    assertNull(graph.depthFirstTraversal(nullV));
```

Comment:
Your `depthFirstTraversal` method iterates over the list returned by `adjacencyLists.get(vertex)`. Because nullV is never added to a graph, it's never added to any `adjacencyLists`, so `adjacencyLists.get(nullV)` will return null. When Java tries to iterate over a null list, it crashes. You could fix it by adding a check to see if that list is null; from there you could handle it with an exception, by returning a null result, or something else.